elasticdl/python/master/k8s_instance_manager.py

```diff
@@ -377,22 +356,32 @@ def _event_cb(self, event):
                     new_worker_id
                 ] = self._worker_pod_priority[worker_id]
             self._start_worker(new_worker_id)
-            with self._lock:
-                self._update_worker_addr(worker_id, new_worker_id)
         elif relaunch_ps:
             logger.info("Relaunching ps.")
             # Note: the ID and service address for relaunched parameter
             # server are intentionally left unchanged to support fault
             # tolerance.
             self._start_ps(ps_id)
 
+        self._worker_addrs = self._get_alive_worker_service__addr()
```

Comment:
Redundant underscore here? Has the new code been tested?

Reply:
The unit test has failed for the reason and I will fix it.